File: app/src/main/java/niks/foreignreader/view/ReaderClickableSpan.java
package niks.foreignreader.view;

import android.graphics.Color;
import android.text.TextPaint;
import android.text.style.ClickableSpan;

public abstract class ReaderClickableSpan extends ClickableSpan {

    protected String mWord;

    public ReaderClickableSpan setWord(String word) {
        mWord = word;
        return this;
    }

    public void updateDrawState(TextPaint textPaint) {
        super.updateDrawState(textPaint);
        textPaint.setUnderlineText(false);
        textPaint.setColor(Color.BLACK);

    }
}

File: app/src/main/java/niks/foreignreader/presentation/reader/ReaderClickableSpan.kt
package niks.foreignreader.presentation.reader

import android.graphics.Color
import android.text.TextPaint
import android.text.style.ClickableSpan

abstract class ReaderClickableSpan : ClickableSpan() {

    protected var word: String = ""

    fun setWord(text: String): ReaderClickableSpan {
        this.word = text
        return this
    }

    override fun updateDrawState(textPaint: TextPaint) {
        super.updateDrawState(textPaint)
        textPaint.isUnderlineText = false
        textPaint.color = Color.BLACK
    }
}
